test/base-testloggerfixture.hpp

```diff
@@ -0,0 +1,94 @@
+/* Icinga 2 | (c) 2025 Icinga GmbH | GPLv2+ */
+
+#ifndef TEST_LOGGER_FIXTURE_H
+#define TEST_LOGGER_FIXTURE_H
+
+#include "base/logger.hpp"
+#include <boost/regex.hpp>
+#include <boost/test/test_tools.hpp>
+#include <BoostTestTargetConfig.h>
+#include <future>
+
+namespace icinga {
+
+class TestLogger : public Logger
+{
+public:
+	DECLARE_PTR_TYPEDEFS(TestLogger);
+
+	struct Expect
+	{
+		std::string pattern;
+		std::promise<bool> prom;
+		std::shared_future<bool> crutch;
```

Comment:
Does this exist only for the case that the timeout expires and because in that case, the entry isn't removed from `m_Expects`?